crates/host/src/wasmbus/mod.rs

```diff
@@ -3651,6 +3803,68 @@ impl Host {
         Ok(ACCEPTED.into())
     }
 
+    #[instrument(skip(self, data))]
+    async fn handle_config_put(
+        &self,
+        entity_id: &str,
+        key: &str,
+        data: Bytes,
+    ) -> anyhow::Result<Bytes> {
+        debug!("handle config entry put");
+        // Simple check for a valid entity key. A valid public nkey is 56 chars and it should start
+        // with the proper character. It isn't worth the overhead of actually parsing the key, but
+        // if that is an issue in the future we can ensure it is a fully parsable nkey.
+        if entity_id.len() != 56 || (!entity_id.starts_with('M') && !entity_id.starts_with('V')) {
+            bail!("Invalid entity ID. The entity ID must be a valid public nkey for an actor or provider");
+        }
+        self.config_data
+            .put(format!("{entity_id}_{key}"), data)
+            .await
+            .context("Unable to store config data")?;
+        // We don't write it into the cached data and instead let the caching thread handle it as we
+        // won't need it immediately.
+
+        self.publish_event("config_set", event::config_set(entity_id, key))
+            .await?;
+
+        Ok(ACCEPTED.into())
+    }
+
+    #[instrument(skip(self))]
+    async fn handle_config_delete(&self, entity_id: &str, key: &str) -> anyhow::Result<Bytes> {
+        debug!("handle config entry deletion");
+
+        self.config_data
+            .purge(format!("{entity_id}_{key}"))
+            .await
+            .context("Unable to delete config data")?;
+
+        self.publish_event("config_deleted", event::config_deleted(entity_id, key))
+            .await?;
+
+        Ok(ACCEPTED.into())
+    }
+
+    #[instrument(skip(self))]
+    async fn handle_config_clear(&self, entity_id: &str) -> anyhow::Result<Bytes> {
+        debug!("handle config deletion");
+
+        let config_cache = self.config_data_cache.read().await;
+        let Some(all_data) = config_cache.get(entity_id) else {
+            return Ok(ACCEPTED.into());
+        };
+        let futs = all_data
+            .keys()
+            .map(|key| self.handle_config_delete(entity_id, key));
+        futures::future::join_all(futs)
+            .await
+            .into_iter()
+            .collect::<Result<Vec<_>, anyhow::Error>>()
+            .context("Unable to delete all config keys. Some keys may remain")?;
```

Comment:
How about https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#method.try_for_each_concurrent instead? It would avoid allocating a vector here and would be a bit easier to read

Reply:
So I did attempt to do this, however, `all_data.keys()` is not a stream, just an iterator. I could wrap it into a stream but it was making it more unreadable, so I opted to use `try_join_all` (which I always forget about). If we're worried about the `Vec` it allocates for results, I can always put everything in a `FuturesUnordered` and then do the stream stuff that way